posthog/celery.py

```diff
@@ -192,6 +194,21 @@ def clickhouse_lag():
             pass
 
 
+@app.task(ignore_result=True)
+def ingestion_lag():
+    from posthog.client import sync_execute
+    from posthog.internal_metrics import gauge
+
+    # Requires https://github.com/PostHog/posthog-heartbeat-plugin to be enabled on team 2
+
+    try:
+        query = """select max(toDateTime(timestamp)) observed_ts, now() now_ts, now() - max(toDateTime(timestamp)) as lag from events where team_id = 2 and event = 'heartbeat';"""
```

Comment:
I mimicked what we have in the clickhouse lag one - potentially for easy copy/pasta verify that the query does sth reasonable

Reply:
Ah, sorry, I misunderstood this, I thought it's an average.
However, this still doesn't seem quite correct. `lag` as calculated here is the amount of time between the last `heartbeat` event entering `EventsProcessor` and the time of this `ingestion_lag` task running. The query makes sense in `clickhouse_lag()`, because we can assume that there's a practically constant stream of events coming into the instance, but filtering by `team_id = 2 AND event = 'heartbeat'` results in something more like a _drip_ of events + for ingestion lag we instead want the time from the event being sent (property `$timestamp`) and the event being available for `SELECT`s (I don't think we have a way to obtain this accurately, but we can just take `_timestamp`, which is the timestamp of the `clickhouse_events_json` topic message, which should be close enough, and `_timestamp` is already used as the starting point in `clickhouse_lag()`).

Reply:
Discussed in slack. 
Michael correctly points out that the metric isn't very accurate due to us running heartbeat every minute, so depending on when this now() is called we will see up to 60s delay there.
However currently that's ok and we're more trying to make sure we know about ingestion being delayed by minutes rather than seconds. Furthermore this will be used for alerting and it would be nice to have a high level alert telling us there's a problem, then we can look at where the problem is (all of which are important to catch): 
(1) ingestion lag 
(2) CH table entry lag 
(3) runEveryMinute not working
(4) plugins are broken
(5) heartbeat plugin is broken - that's important because then our monitoring is broken